scene/gui/label.cpp

```diff
@@ -585,7 +588,7 @@ void Label::set_percent_visible(float p_percent) {
 
 	} else {
 
-		visible_chars = get_total_character_count() * p_percent;
+		visible_chars = get_page_character_count() * p_percent;
```

Comment:
Potential problem here: If the label is resized after setting `percent_visible`, it will fail to update the visible characters to the new amount, no?

Reply:
You are right, in this case, I call the `set_percent_visible` after the label is updated. Even in the current code, I have to make the label `visible = false` then `visible = true` to get the correct value of some properties.

Here some example of my current code in godot script, just to reset the label size. Without this the label size is incorrect.

```
chattext.text = "."
chattext.visible = false
chattext.visible = true
chattext.autowrap = false
chattext.rect_min_size = Vector2(0, 0)
chattext.rect_size = Vector2(40, 20)
chattext.text = text
chattext.visible = false
chattext.visible = true
		
if chattext.rect_size.x > global.chat_max_width:
  chattext.autowrap = true
  chattext.rect_min_size.x = global.chat_max_width
  chattext.visible = false
  chattext.visible = true

chatbox.rect_size = Vector2(0, 0)

chattext.percent_visible = 0
chattext.lines_skipped = 0
```

If you look at this code, I set the visibility of `chattext` to `false` then `true` several times just to redraw it correctly.
I`ve tried several times, this is only the working code at the moment.